Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -14,12 +15,89 @@
 #include <SDL3/SDL.h>
 
 #include "Common/Event.h"
+#include "Common/Keyboard.h"
 #include "Common/Logging/Log.h"
 #include "Common/ScopeGuard.h"
+#include "Core/Host.h"
 
 #include "InputCommon/ControllerInterface/ControllerInterface.h"
 #include "InputCommon/ControllerInterface/SDL/SDLGamepad.h"
 
+namespace
+{
+using UniqueSDLWindow = std::unique_ptr<SDL_Window, void (*)(SDL_Window*)>;
+
+// Based on sdl2-compat 76eb981a4c376bcaf615c0af37d46512ba45cfb8
+SDL_Window* SDL_CreateWindowFrom(void* handle)
+{
+  SDL_PropertiesID props = SDL_CreateProperties();
+  if (!props)
+    return nullptr;
+
+  if (const char* hint = SDL_GetHint("SDL_VIDEO_WINDOW_SHARE_PIXEL_FORMAT"); hint)
+  {
+    // This hint is a pointer (in string form) of the address of
+    // the window to share a pixel format with
+    SDL_Window* other_window = nullptr;
+    if (SDL_sscanf(hint, "%p", &other_window))
+    {
+      void* ptr = SDL_GetPointerProperty(SDL_GetWindowProperties(other_window),
+                                         SDL_PROP_WINDOW_WIN32_HWND_POINTER, nullptr);
+      SDL_SetPointerProperty(props, SDL_PROP_WINDOW_CREATE_WIN32_PIXEL_FORMAT_HWND_POINTER, ptr);
+      SDL_SetBooleanProperty(props, SDL_PROP_WINDOW_CREATE_OPENGL_BOOLEAN, true);
+    }
+  }
+
+  if (SDL_GetHintBoolean("SDL_VIDEO_FOREIGN_WINDOW_OPENGL", false))
+    SDL_SetBooleanProperty(props, SDL_PROP_WINDOW_CREATE_OPENGL_BOOLEAN, true);
+
+  if (SDL_GetHintBoolean("SDL_VIDEO_FOREIGN_WINDOW_VULKAN", false))
+    SDL_SetBooleanProperty(props, SDL_PROP_WINDOW_CREATE_VULKAN_BOOLEAN, true);
+
+  SDL_SetPointerProperty(props, "sdl2-compat.external_window", handle);
+  SDL_Window* window = SDL_CreateWindowWithProperties(props);
```

Comment:
I get a segfault here inside `wl_proxy_get_version`. I'm using Wayland.

Reply:
Very odd that it didn't happen on my end. Could you please provide more information:
 - Linux distribution name and version
 - Wayland version
 - SDL version (and if it's the one bundled with Dolphin)
 - Was the build manual or the buildbot's flatpak

A Dolphin crash for this kinda of feature is quite bad. So at the very least, I'd like to reproduce it and find a way to fix it or prevent it from happening.

I've seen PR/issues with crashes involving `wl_proxy_get_version` on SDL's GitHub repository but I don't know if it's related. And even if it's, I should at the very least provide a workaround so it doesn't crash Dolphin.

Reply:
Arch Linux
Wayland 1.23.1-2 from Arch repos
System sdl3-3.2.16-1 from Arch repos
I'm building your PR myself.